src/include/ipc/topology.h

```diff
@@ -47,6 +47,7 @@ enum sof_comp_type {
 	SOF_COMP_ASRC,		/**< Asynchronous sample rate converter */
 	SOF_COMP_DCBLOCK,
 	SOF_COMP_SMART_AMP,		/**< smart amplifier component */
+	SOF_COMP_POST_PROCESS,
```

Comment:
Shall we just call this PROCESS now that we are using UUID. @plbossart ? This would cover all cases.

Reply:
Right, I thought we'd stop adding such types.

Reply:
Wait a second, so should I refrain from adding a new type to sof_comp_type and use the new UUID approach for components right? (I need to revise the UUID PR again).

Also, what about the "subtypes" (flavours) of PROCESS component like (mp3, aac, etc). Can I still use  enum sof_ipc_process_type for that, right?

Reply:
@dbaluta the intention is to move away from kernel updates to minimise upstreaming (since the kernel does not care about general processing components). Having said that the ALSA compressed PCM may need to know the media type or does it not care and just acts as a passthrough ? @plbossart can you comment, does compressed ALSA API need to know media types ?

Reply:
It only needs to know enough to translate the userspace request to properly select the correct type. I think a good idea is to just have the enum copied and the value of the type field outright passed through directly to the firmware.

Reply:
@dbaluta when ALSA compressed device is created, ALSA needs to know the media type. But the PROCESS component on the FW side is created much early when the topology is parsed. Current infrastructure allows to set both component type (PROCESS) and process component type (mp3/aac etc) when the topology is created.

See sof_ipc_comp_process.comp.type and sof_ipc_comp_process.type.
```
/* generic "effect", "codec" or proprietary processing component */
struct sof_ipc_comp_process {
	struct sof_ipc_comp comp;
	struct sof_ipc_comp_config config;
	uint32_t size;	/**< size of bespoke data section in bytes */
	uint32_t type;	/**< sof_ipc_process_type */

	/* reserved for future use */
	uint32_t reserved[7];

	unsigned char data[0];
} __attribute__((packed));
```
I think for the first implementation we will use by default:
*  sof_ipc_comp_process.comp.type = SOF_COMP_PRCESS  (this is read from topology file)
*  sof_ipc_comp_process.type = SOF_PROCESS_CODEC_MP3 (this is harcoded) (1)

(1) ideally will be received by ALSA Compress API from userspace and will be set when we start a stream.

Reply:
@dbaluta this is an interesting one: i think the compressed API lets you specify the codec type at run time, when setting the params. So in theory it's possible to have a generic decoder that can handle multiple formats. Setting a unique format at the topology creation time is possible, but would be a restriction (just like setting a single frequency for a PCM device is permitted but isn't always required).